addons/web/static/src/views/fields/integer/integer_field.js

```diff
@@ -51,7 +51,12 @@ export class IntegerField extends Component {
             !this.props.formatNumber ||
             (!this.props.readonly && this.props.inputType === "number")
         ) {
-            return this.value;
+            // Basic number checking to make sure we send out a number
+            // not a Boolean value in Integer field.
+            if (isNumber(this.value)) {
+                return this.value;
+            }
+            return "";
```

Comment:
Why not simply checking if `value === false` as we do in `formatInteger`? 🤔 

It does not seem need to create a function to check if the value is a number.

```suggestion
            if (this.value === false) {
                return "";
            }
            return this.value;
```

Reply:
Yeah I agree but the broader idea was to replace the repetitive check in formatFloat, formatInteger to have a basic check but It is not viable for our case